pipe.go

```diff
@@ -6,58 +6,58 @@ import (
 
 // Pipe allows messages to flow through the processors.
 type Pipe interface {
+	// Queue gets the queued Messages for each Node.
+	Queue() []NodeMessage
+	// Forward queues the data to all processor children in the topology.
 	Forward(*Message) error
+	// Forward queues the data to the the given processor(s) child in the topology.
 	ForwardToChild(*Message, int) error
+	// Commit commits the current state in the sources.
 	Commit(*Message) error
 }
 
 // ProcessorPipe represents the pipe for processors.
 type ProcessorPipe struct {
-	currentNode Node
+	node  Node
+	queue []NodeMessage
 }
 
 // NewProcessorPipe create a new ProcessorPipe instance.
-func NewProcessorPipe() *ProcessorPipe {
-	return &ProcessorPipe{}
+func NewProcessorPipe(node Node) *ProcessorPipe {
+	return &ProcessorPipe{
+		node:  node,
+		queue: []NodeMessage{},
+	}
 }
 
-// SetNode sets the topology node that is being processed.
+// Queue gets the queued Messages for each Node.
 //
-// The is only needed by the task and should not be used
-// directly. Doing so can have some unexpected results.
-func (p *ProcessorPipe) SetNode(n Node) {
-	p.currentNode = n
+// Reading the node message queue will reset the queue.
+func (p *ProcessorPipe) Queue() []NodeMessage {
```

Comment:
Do we really need the `Queue` method exposed, especially if it has such a dangerous side-effect?
Also, because of this side-effect, the method should be called `PopQueue()` or `ConsumeQueue()`

Reply:
Generally I agree, but for this purpose, we unfortunately cannot. I have plans to get rid of this in a future release.